Admission scenarios screen (EXPOSUREAPP-11751)

mtwalli opened the pull request:
## Changes 
- Admission scenarios screen 
- Update admission scenarios and selectedScenario Id after selection
- Trigger DccWalletInfo calculation
- Suppress Error per person calculation to continue for other persons
- Screenshots
- With `2.18` bug fix , UpdateTask became kind of Obsolete -> 🗑️ 
- Reaming 
```diff
-  suspend fun triggerCalculationAfterConfigChange(
+  suspend fun triggerAfterConfigChange(

- suspend fun triggerCalculationAfterCertificateChange(
+ suspend fun triggerNow(
```

## TODO
- [x] unit tests 

## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11751

mtwalli committed:
Admission scenario tile

mtwalli committed:
Navigate to Admission Scenarios screen

mtwalli committed:
ViewModel

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11752-admission-tile

mtwalli committed:
Merge branch 'feature/11752-admission-tile' of https://github.com/corona-warn-app/cwa-app-android into feature/11752-admission-tile

mtwalli committed:
Fix log

mtwalli committed:
Update AdmissionScenariosViewModel.kt

mtwalli committed:
Expose scenarios

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11752-admission-tile

mtwalli committed:
Merge branch 'feature/11752-admission-tile' of https://github.com/corona-warn-app/cwa-app-android into feature/11752-admission-tile

mtwalli committed:
Display tile `CCLText`

mtwalli committed:
lint

mtwalli committed:
Update PersonOverviewViewModelTest.kt

mtwalli committed:
Shared VM

mtwalli committed:
Lint

mtwalli committed:
Handle button click

mtwalli committed:
AppBar theme

mtwalli committed:
Format

mtwalli committed:
Transition

mtwalli committed:
Admission state

mtwalli committed:
Selected item

mtwalli committed:
Update PersonOverviewViewModelTest.kt

mtwalli committed:
Update Admission tile based on changes

mtwalli committed:
Tests

mtwalli committed:
UI tweaking / Figma

mtwalli committed:
Address TODOs

mtwalli committed:
Correct name

mtwalli committed:
Adjust margins

mtwalli committed:
Merge branch 'feature/11752-admission-tile' into feature/11751-admission-scenarios-fragment

mtwalli committed:
lint

mtwalli committed:
Disabled item colour

mtwalli committed:
Tile provider

mtwalli committed:
Remove

mtwalli committed:
lint

mtwalli committed:
Revert "Remove"

This reverts commit 7f736337c14fb40cc9778463a303a83d14c09183.

mtwalli committed:
Update AdmissionTileProviderTest.kt

mtwalli committed:
Update AdmissionTileProviderTest.kt

mtwalli committed:
Line / alignment with certificates card

mtwalli committed:
Alignement

mtwalli committed:
tile background colour

mtwalli committed:
Merge branch 'feature/11752-admission-tile' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Dark mode colours

mtwalli committed:
Update DccAdmissionCheckScenariosTest.kt

mtwalli committed:
Calculating dialog

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Lint

mtwalli committed:
Fix colors

mtwalli committed:
Fix merge errors

mtwalli committed:
Rename

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Trigger calculations

mtwalli committed:
Fix merge errors

mtwalli committed:
lint

mtwalli committed:
Fix test

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
default admission scenario id

mtwalli committed:
Navigate back on error

mtwalli committed:
Catch error per person

mtwalli committed:
Don't show calculation error

mtwalli committed:
Refactoring

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

# Conflicts:
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModel.kt

mtwalli committed:
Update AdmissionScenariosViewModel.kt

mtwalli committed:
Update DccWalletInfoUpdateTrigger.kt

mtwalli committed:
Trigger Now

mtwalli committed:
Trigger now now

mtwalli committed:
Adjust item color

mtwalli committed:
Lint

mtwalli committed:
Screenshots tests

mtwalli committed:
Fix screenshots

mtwalli committed:
Revert renaming

mtwalli committed:
Revert renaming

mtwalli committed:
UI tweaking

mtwalli committed:
Screenshots

mtwalli committed:
Item txt color

mtwalli committed:
Open for testing

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Log `d` handled error

KathrinLuecke reviewed (dismissed):
UA approved

chiljamgossow reviewed (dismissed):
Looks good to, tested it
The popup seems to be dismissed right away as the calculation is fast. Maybe we can drop it if displaying it is slower than the calculation, but that would be an enhancement

mtwalli committed:
Create AdmissionScenariosViewModelTest.kt

mtwalli commented:
> Looks good to, tested it
> The popup seems to be dismissed right away as the calculation is fast. Maybe we can drop it if displaying it is slower than the calculation, but that would be an enhancement

That is right for a few person certificates ,but in case there are many persons we need the indicator and maybe in the future Configuration will change and do more work. let us keep as in Figma

mtwalli committed:
Correct logs

mtwalli committed:
Update AdmissionScenariosViewModelTest.kt

mtwalli committed:
Update AdmissionScenariosViewModelTest.kt

mtwalli committed:
Delete AdmissionSharedViewModel.kt

mtwalli committed:
Update PersonOverviewViewModelTest.kt

mtwalli committed:
Update PersonOverviewViewModelTest.kt

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

# Conflicts:
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt
#	Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModel.kt

mtwalli committed:
Fix merge error

mtwalli committed:
Catch errors

mtwalli committed:
Refactoring

After 2.18 changes to use flow Task became kind of obsolete and its running types

mtwalli committed:
Update name

mtwalli committed:
Update DccWalletInfoUpdateTriggerTest.kt

mtwalli committed:
Merge branch 'release/2.19.x' into feature/11751-admission-scenarios-fragment

mtwalli committed:
Lint

mtwalli committed:
Cover admission Id

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt:
```diff
@@ -22,66 +21,65 @@ import javax.inject.Singleton
-                    admissionScenarioId = getAdmissionScenarioId()
-                ),
-                originTag = TAG
-            )
```
Task removed

LukasLechnerDev commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewViewModelTest.kt:
```diff
@@ -259,6 +260,28 @@ class PersonOverviewViewModelTest : BaseTest() {
     }
 
+    @Test
+    fun `openAdmissionScenarioScreen - success`() {
```
These 2 tests are fine. However, I like it a bit better if we would call `triggerNow()` in these tests and verify if `dccWalletInfoCalculationManager.triggerNow()` was called and the correct scenarioId was passed. Then we wouldn't need "@VisibleForTesting" for `admissionScenarioId()` .

mtwalli replied:
this function is not triggering DccWalletInfo ,it calculates the scenarios, but I will verify that :)

mtwalli replied:
updated tests

LukasLechnerDev replied:
Thanks!

LukasLechnerDev commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/ui/admission/AdmissionScenariosViewModelTest.kt:
```diff
@@ -0,0 +1,95 @@
+    @Test
+    fun selectScenario() {
+        instance().selectScenario("admissionScenarioId")
+        coVerifySequence {
```
Maybe also verify here if the correct events were posted? (`Calculation` and `CalculationDone`)?

mtwalli replied:
I tried ,but SingleLiveData with two emissions is really flaky in this test , no clue what is wrong

LukasLechnerDev replied:
Ok. Then let's leave it as is.

LukasLechnerDev reviewed (dismissed):
Great work! 2 optional comments.

mtwalli committed:
Verify

mtwalli committed:
Specific Ids

mtwalli committed:
Provide dispatcher

mtwalli committed:
Update AdmissionScenariosViewModelTest.kt

mtwalli committed:
lint
